apiservers/engine/backends/image.go

```diff
@@ -107,36 +107,28 @@ func (i *Image) Images(filterArgs string, filter string, all bool) ([]*types.Ima
 		v1Map[image.ID] = v1Image
 	}
 
-	for id, v1Image := range v1Map {
-
-		size := getImageSize(v1Map, id)
-
+	for _, v1Image := range v1Map {
 		newImage := &types.Image{
 			ID:       v1Image.ID,
 			ParentID: v1Image.Parent,
 			RepoTags: []string{"<fixme>:<fixme>"}, // TODO(jzt): replace with actual tags
 			//RepoDigests: []string{"<fixme>:<fixme>"},
-			Created:     v1Image.Created.Unix(),
-			VirtualSize: size,
-			Labels:      v1Image.Config.Labels,
+			Created: v1Image.Created.Unix(),
+			Labels:  v1Image.Config.Labels,
 		}
 		result = append(result, newImage)
 	}
 
 	sort.Sort(sort.Reverse(byCreated(result)))
 
-	return result, nil
-}
-
-// add size of all parent images plus this image size
-func getImageSize(v1Map map[string]*v1.V1Image, id string) int64 {
-	var s int64
-	parentID := v1Map[id].Parent
-	for parentID != "" {
-		s += v1Map[parentID].Size
-		parentID = v1Map[parentID].Parent
+	var size int64
+	for i := len(result) - 1; i >= 0; i-- {
+		image := result[i]
+		size += v1Map[image.ID].Size
```

Comment:
You already know a running sum when you have multiple images won't work.  You can fix that by
v1Map[image.ID].Size += v1Map[image.Parent].Size.